test/langtools/tools/javac/launcher/MultiFileSourceLauncherTests.java

```diff
@@ -75,7 +75,7 @@ void testLoadingOfEnclosedTypes(@TempDir Path base) throws Exception {
                     public class Hello {
                       public static void main(String... args) throws Exception {
                         System.out.println(Class.forName("World$Core"));
-                        System.out.println(Class.forName("p.q.Unit$First$Second"));
+                        System.out.println(Class.forName("p.q.Unit$123$Fir$t$$econd"));
```

Comment:
I think I would suggest two more tests:
- one that would (in a separate run) lookup `p.q.Unit$123$Fir$t` (to ensure the search is broad enough)
- one with a package name with `$`.

Reply:
Added both tests a suggested.

The first is not test in the separate run, but a separate top-level compilation unit.